### ISSUE TITLE
Maximum execution time of 30 seconds exceeded when adding a PDF document

### ISSUE THREAD

**Issue by Massedil** [wallabag/wallabag#7667]
### Environment

* **Version**: 2.6.9 (latest)
* **Installation**: downloading the package
* **PHP version**: PHP 7.4.33
* **OS**: Debian GNU/Linux 11 (bullseye)
* **Database**: SQLite
* **Parameters**:

<details>
  <summary>My <code>app/config/parameters.yml</code> is:</summary>

  ```
  PASTE HERE
  ```
</details>

### What steps will reproduce the bug?

Try to add this article (heavy PDF) : 
https://dumas.ccsd.cnrs.fr/dumas-02289223v1/preview/1892-18adavos_Davos_Amandine.pdf

It fails with : 

```
PHP message: PHP Fatal error:  Maximum execution time of 30 seconds exceeded in /var/www/wallabag/latest/vendor/smalot/pdfparser/src/Smalot/PdfParser/RawData/RawDataParser.php on line 744
```

### Side effects

The wallabag app (https://github.com/wallabag/android-app) receive a HTTP 500 error, but retries to add this PDF again and again. The user can't stop it and can't use the app anymore.

### Help

How can I disable the PDF parser before this issue is solved ?

**Comment by j0k3r** [comment]
You can't disable the PDF parser.

You can try to increase the maximum execution time of PHP on your server.

I don't know the android app, but maybe you can remove the entries from there to avoid infinite retries?

**Comment by Massedil** [comment]
Thanks for your quick reply !

Can you reproduce the issue by adding the PDF on your own wallabag instance ?

> You can try to increase the maximum execution time of PHP on your server.

Yes, but more than 30 seconds to add a link is not normal, don't you agree ?

> I don't know the android app, but maybe you can remove the entries from there to avoid infinite retries?

Except if I'm wrong, the entry is not visible inside the app, so the user can't delete it. I will report the problem directly to their github.

**Comment by Massedil** [comment]
> Maybe you own server isn't fast enough to handle parsing big PDF?

Yes, it seems my server is less performant than yours. But I imagine with a bigger PDF, you will have the same problem on your server also.

What do you think about a `max downloadable URL size` option that abort file content recovery and just add the article inside wallabag and a content like : "This content is too heavy to be saved" ?

**Comment by j0k3r** [comment]
That could be an interesting workaround.
